File: src/main/java/com/junchen/back/service/AmRulesService.java
package com.junchen.back.service;
import java.util.List;
import com.junchen.back.pojo.ErpAmRules;

import com.junchen.back.myentity.PageResult;
/**
 * amRules服务层接口
 * @author Administrator
 *
 */
public interface AmRulesService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<ErpAmRules> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(ErpAmRules amRules);
	
	
	/**
	 * 修改
	 */
	public void update(ErpAmRules amRules);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public ErpAmRules findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(ErpAmRules amRules, int pageNum,int pageSize);

	/**
	 * 查找出发表过规章制度的员工的id
	 * @return
	 */
	public List<ErpAmRules> findByStaffId();
	
}

File: src/main/webapp/js/service/amRulesService.js
//amRules服务层
app.service('amRulesService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../amRules/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../amRules/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../amRules/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../amRules/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../amRules/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../amRules/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../amRules/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
	
	this.findStaff = function(){
		return $http.get('../amRules/findByStaffId.do');
	}
	
});
